src/main/java/com/example/onlinebookstore/service/impl/CartServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+import com.example.onlinebookstore.mapper.ShoppingCartMapper;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.cart.item.CartItemRepository;
+import com.example.onlinebookstore.repository.user.UserRepository;
+import com.example.onlinebookstore.service.CartService;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CartServiceImpl implements CartService {
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    public ShoppingCartResponseDto getShoppingCart(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        return shoppingCartMapper.toDto(user.getShoppingCart());
+    }
+
+    @Override
+    public CartItemResponseDto saveCartItem(Authentication authentication,
+                                            CreateCartItemDto createCartItemDto) {
+        User user = (User) authentication.getPrincipal();
+        CartItem cartItem = cartItemMapper.toModel(createCartItemDto);
+        cartItem.setShoppingCart(user.getShoppingCart());
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public CartItemResponseDto updateCartItem(Authentication authentication,
+                                              UpdateCartItemDto updateCartItemDto, Long id) {
+        User user = (User) authentication.getPrincipal();
+        CartItem cartItem = cartItemRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Cart item not found by id: " + id));
+        if (!Objects.equals(cartItem.getShoppingCart().getUser().getId(), user.getId())) {
+            throw new EntityNotFoundException("Cart item not found by id: " + id);
+        }
+        cartItem.setAmount(updateCartItemDto.getAmount());
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public void deleteCartItem(Authentication authentication, Long id) {
+        User user = (User) authentication.getPrincipal();
+        CartItem cartItem = cartItemRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Cart item not found by id: " + id));
+        if (!Objects.equals(cartItem.getShoppingCart().getUser().getId(), user.getId())) {
+            throw new EntityNotFoundException("Cart item not found by id: " + id);
+        }
+        if (cartItem.isDeleted()) {
+            throw new DeleteEntityException("This cart item is already deleted with id: " + id);
+        }
```

Comment:
Please just add proper annotation on entity class not to get soft-deleted items.
like @Where(clause = "is_deleted = FALSE")

And remove explicit checks in code